modules/fedimint-wallet-server/src/lib.rs

```diff
@@ -329,51 +321,100 @@ impl ServerModule for Wallet {
+            items.push(WalletConsensusItem::Feerate(fee_rate_proposal));
+        }
+
+        ConsensusProposal::new_auto_trigger(items)
     }
 
     async fn begin_consensus_epoch<'a, 'b>(
         &'a self,
         dbtx: &mut ModuleDatabaseTransaction<'b>,
         consensus_items: Vec<(PeerId, WalletConsensusItem)>,
-        consensus_peers: &BTreeSet<PeerId>,
+        _consensus_peers: &BTreeSet<PeerId>,
     ) -> Vec<PeerId> {
         trace!(?consensus_items, "Received consensus proposals");
 
-        // Separate round consensus items from signatures for peg-out tx. While
-        // signatures can be processed separately, all round consensus items
-        // need to be available at once.
-        let UnzipWalletConsensusItem {
-            peg_out_signature: peg_out_signatures,
-            round_consensus: round_items,
-        } = consensus_items.into_iter().unzip_wallet_consensus_item();
+        for (peer_id, item) in consensus_items {
+            match item {
+                WalletConsensusItem::BlockHeight(block_height) => {
+                    let current_vote = dbtx
+                        .get_value(&BlockHeightVoteKey(peer_id))
+                        .await
+                        .unwrap_or(0);
 
-        // Save signatures to the database
-        self.save_peg_out_signatures(dbtx, peg_out_signatures).await;
+                    // consensus block height can never decrease
+                    match block_height.cmp(&current_vote) {
+                        Ordering::Greater => {
+                            let old_consensus_height = self.consensus_block_height(dbtx).await;
 
-        let last_height = self.consensus_height(dbtx).await.unwrap_or(0);
+                            dbtx.insert_entry(&BlockHeightVoteKey(peer_id), &block_height)
+                                .await;
 
-        match Self::round_consensus(last_height, round_items, consensus_peers) {
-            Ok(round_consensus) => {
-                self.sync_up_to_consensus_height(dbtx, round_consensus.block_height)
-                    .await;
+                            let new_consensus_height = self.consensus_block_height(dbtx).await;
 
-                dbtx.insert_entry(&RoundConsensusKey, &round_consensus)
-                    .await;
-                vec![]
+                            if new_consensus_height > old_consensus_height {
+                                self.sync_up_to_consensus_height(
+                                    dbtx,
+                                    old_consensus_height,
+                                    new_consensus_height,
+                                )
+                                .await;
+                            }
+                        }
+                        Ordering::Less => {
+                            warn!("Peer block height vote decreased");
+                        }
+                        Ordering::Equal => {}
```

Comment:
Can this happen with correct peers?

Reply:
Less could not happen. If we filter redundant items within a session equal may happen if submitted in two different sessions by a valid peer.